client/asset/dcr/dcr.go

```diff
@@ -1147,17 +1376,17 @@ func (dcr *ExchangeWallet) maxOrder(lotSize, feeSuggestion, maxFeeRate uint64) (
 	if err != nil {
 		return nil, nil, fmt.Errorf("error parsing unspent outputs: %w", err)
 	}
-	var avail uint64
-	for _, utxo := range utxos {
-		avail += toAtoms(utxo.rpc.Amount)
-	}
+	avail := sumUTXOs(utxos)
 
 	// Start by attempting max lots with a basic fee.
 	basicFee := dexdcr.InitTxSize * maxFeeRate
 	lots := avail / (lotSize + basicFee)
-	cfg := dcr.config()
+	// NOTE: Split tx is an order-time option. The max order is generally
+	// attainable when split is used, regardless of whether they choose it on
+	// the order form. Allow the split for max order purposes.
+	trySplitTx := true
```

Comment:
Yeah, it's more nuanced than I let on, but this is referring to when there are reserves to respect, because the split maximizes the usable balance by allowing the split tx change to offset reserves.

However, the split actually eats a bit extra in fees because of the split tx baggage, but because `dcr.estimateSwap` falls back to skipping the split if needed, we can be sure that `maxOrder` calling `estimateSwap` with `trySplitTx=true` will give the highest estimate one way or another.  The comments in the revised tests in https://github.com/decred/dcrdex/pull/2103/commits/57db9ad43b8cb16a8bbc2fd0047c3bb2b2dd58d2 demonstrate this I think.